🚍 Stops/Station/TNM | CR | Cross out skipped/cancelled stops

thecristen opened the pull request:
#### Summary of changes
**Asana Ticket:** [🚍 Stops/Station/TNM | CR | Cross out skipped/cancelled stops](https://app.asana.com/0/385363666817452/1175248604991146/f)

* Adds `schedule_relationship` to the predictions used on stop pages and transit near me
* Shows predictions with `schedule_relationship` of skipped or canceled as "Canceled" with strikethrough
* Shows the absence of a prediction as "Scheduled" instead of "On time". 
* Adjusts the `$text-muted` color to use a darker gray to ensure good contrast.

Stops

![image](https://user-images.githubusercontent.com/2136286/87698058-f508bd80-c760-11ea-9840-43fae1128ea0.png)


Transit Near Me

![image](https://user-images.githubusercontent.com/2136286/87697996-e1f5ed80-c760-11ea-8490-d7a6d2a92763.png)


---

Before getting review, please check the following:

* [x] Does frontend functionality render and work correctly in IE?
* [x] Have we load-tested any new pages or internal API endpoints that will receive significant traffic?
* [x] Are interactive elements accessible to screen readers?
* [x] Have you checked for tech debt you can address in the area you're working in?
* [x] If this change involves routes, does it work correctly with pertinent "unusual" routes such as the combined Green Line, Silver Line, Foxboro commuter rail, and single-direction bus routes like the 170?
* [x] Are the changes organized into self-contained commits with descriptive and well-formatted commit messages?

ingridpierre commented:
How hard would it be to change the TNM logic for CR to only show one schedule at a time as part of this ticket? Showing two on a row is a regression similar to (but different from): https://app.asana.com/0/385363666817452/1184312880988849/f 

Notably, Train 512 is not the train number for both the listed scheduled trains here:
![image](https://user-images.githubusercontent.com/5806643/87816556-88abbe00-c835-11ea-8e8b-c289fb221152.png)

arkadyan commented on apps/site/assets/ts/components/Headsign.tsx:
```diff
@@ -45,22 +45,29 @@ const renderTrainName = (trainName: string): ReactElement<HTMLElement> => (
+      {timeForCommuterRail(
+        data,
+        `m-tnm-sidebar__time-number ${
+          status === "Canceled" ? "strikethrough" : ""
```
I think a quick test for this visual behavior could be useful to prevent regressions.

thecristen commented:
Added the tests and adjusted the TNM logic for CR to only show one schedule at a time.
Saw some more incorrect CR train numbers this morning (including even a `Train null`!), so will file a new ticket for that.

ingridpierre commented:
Currently I'm seeing some unexpected displays here.

**Two disconnected train times, stacked on a departed status:**
![image](https://user-images.githubusercontent.com/5806643/88065100-2b1bb800-cb3a-11ea-9bdb-4855c9af12ff.png)

----------
**Possible regression on late train styling?**
CURRENT 
Late trains >5 mins late are not appearing styled as they once were. Here is what I saw:
![image](https://user-images.githubusercontent.com/5806643/88065287-68804580-cb3a-11ea-93b9-c2eaac468769.png)
![image](https://user-images.githubusercontent.com/5806643/88065425-91a0d600-cb3a-11ea-982d-c966c0bd985b.png)

**DESIRED** appearance of late trains;
![image](https://user-images.githubusercontent.com/5806643/88067346-de85ac00-cb3c-11ea-8ba7-dbba8b8bb4e1.png)

thecristen commented:
> Two disconnected train times, stacked on a departed status

This was actually a visual regression, thanks for catching it! I am fixing this in a moment.

> Late train styling

I'm not sure that's a regression -- how does it compare to what's shown on master? Our commuter status doesn't have any "Late" styling as far as I can tell, so maybe we'll have to add that (in a future ticket).

ingridpierre commented:
Looks like it is a regression, but I'm not sure how long it has been happening. On this ticket from the original TNM project we have delayed status styled differently in the route cards: https://app.asana.com/0/961607479498016/1100182096176852/f

**Delayed/Early**

- show the original scheduled time with a strikethrough
- show the new time
- show the Delayed/Early train status


--------


There is definitely a regression on the line diagram. Here's an old screenshot of production many months ago.
![image](https://user-images.githubusercontent.com/5806643/88099976-f32c6900-cb69-11ea-815c-5a52002aee21.png)

ingridpierre commented:
@thecristen Sorry for the goose chase on statuses/delayed/Late trips.

I'm still not able to see a canceled trip, but I do like the styling so far. The grey we're using on the 'Scheduled', but not predicted trips is a bit different than the other places we're using grey for schedules in the cards. Compare the grey for this scheduled bus:

![image](https://user-images.githubusercontent.com/5806643/88111186-7904e000-cb7b-11ea-8ce3-b3e981cedf32.png)
![image](https://user-images.githubusercontent.com/5806643/88111226-8a4dec80-cb7b-11ea-8475-655278775d89.png)

thecristen commented:
@ingridpierre I see!
Looking at that scheduled bus time, I'm finding it has insufficient contrast (3.95 or 4.01 depending on which of my two tools you ask..) – so I am going to adjust it to use `$gray` instead of `$gray-light` (which will also make it match the 'Scheduled').
